### PR TITLE
fix: wasm error during transaction

### LINKED ISSUE #243: Unsuccessful Transactions Due to Bad Signature and Trap Error
When connected to [Polkadot Wallet Snap Example dApp](https://polkadot.snap.chainsafe.io/) and attempting a simple transaction, I encountered the following error in the console:

> MetaMask - RPC Error: 4003: Client error: Execution failed: Execution aborted due to trap: wasm trap: wasm `unreachable` instruction executed

When I use the [Staking Dashboard](https://staking.polkadot.cloud/), the operation is canceled, and I receive the following error in the console:

> ExtrinsicStatus:: 1010: Invalid Transaction: Transaction has a bad signature

Perhaps it is caused by the runtime upgrade to v1.2.5?
- [Runtimes 1.2.5](https://github.com/polkadot-fellows/runtimes/releases/tag/v1.2.5)
- [PR 337](https://github.com/polkadot-fellows/runtimes/pull/337)

Chrome 127.0.6533.100
Metamask 12.0.1
ChainSafe Snap 0.11.0

Note: PolkaGate snap does work.

### PR DESCRIPTION
**Pull request by Lykhoyda**
Closes #243 
- polkadot libraries were updated
- updates for Metamask cli and build 
- switched to node 18 (16 got deprecated with the new metamask cli)
- GitHub actions uses v 18 of node 
- Removed shasum check on GitHub actions as a new CLI updates the manifest file if it doesn't match

### CODE REVIEW

File: packages/snap/src/util/validation.ts [hunk 1]
```diff
@@ -18,6 +18,8 @@ const SignaturePayloadJSONSchema = type({
   version: number()
 });
 
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (Lykhoyda) [inline packages/snap/src/util/validation.ts:21]: in progress to remove the comments

File: .github/workflows/ci-workflow.yaml [hunk 2]
```diff
@@ -18,25 +18,13 @@ jobs:
...
-        id: verify-changed-files
-        with:
-          files: |
-            packages/snap/snap.manifest.json
-      - name: Check if snap manifest updated
-        if: steps.verify-changed-files.outputs.files_changed == 'true'
-        run: |
-          echo "Snap manifest not updated: ${{ steps.verify-changed-files.outputs.changed_files }} 
-          or @metamask/snaps-cli version is different from local version"
-          exit 1
```
[the remark this answers is not in the document]
Comment (Lykhoyda) [reply .github/workflows/ci-workflow.yaml:39]: The CLI will fix the shasum automatically during the build if it doesn't match. So probably it's not necessary anymore. Do you have any other scenario in your mind? 
```md
"snap.manifest.json" "shasum" field does not match computed shasum. Got "DD1EgPFXRVTdFGPbx7Qcch1bdC5jeTj+13TPrztmgQk=", expected "4ZnZF8T0dPgKPTW3dRIee+Fww7MZKEobF0BO7iz+K4M=". (fixed)
```

File: packages/snap/package.json [hunk 3]
```diff
@@ -25,7 +25,7 @@
     "prepublish": "yarn run build && mkdir ipfs && cp package.json ipfs/package.json",
     "publish": "export BUNDLE_URL=`ipd --no-clipboard -O -p infura dist/bundle.js` && sed -i 's/http:\\/\\/localhost:8081\\/dist\\/bundle.js/https:\\/\\/ipfs.infura.io\\/ipfs\\/'\"$BUNDLE_URL\"'/g' ipfs/package.json && ipd --no-clipboard -O -p infura ipfs/package.json",
     "postpublish": "rm -rf ipfs",
-    "serve": "mm-snap serve",
+    "serve": "mm-snap serve & http-server",
```
[the remark this answers is not in the document]
Comment (Lykhoyda) [reply packages/snap/package.json:28]: The build configuration has changed a bit, and the command mm-snap serves only snap. To keep our index.html example I added http-server to serve the page.

File: .github/workflows/ci-workflow.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (BeroBurny) [reply .github/workflows/ci-workflow.yaml:39]: We had issues while multiple people worked on multiple parts at the same time, and this helped to reduce confusion and false positives

File: packages/snap/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (BeroBurny) [reply packages/snap/package.json:28]: that `index.html` should be deprecated and removed, we have an example package for it

File: .github/workflows/ci-workflow.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lykhoyda) [reply .github/workflows/ci-workflow.yaml]: @BeroBurny for some reason the shasum is inconsistent between build on CI and locally. It's probably an issue with cached dependencies on github actions.

[the remark this answers is not in the document]
Comment (BeroBurny) [reply .github/workflows/ci-workflow.yaml]: that can be an issue with globally installed snap cli

**Review by BeroBurny — APPROVED** [review]

**Pull request merged**